docs/integrations/consul/k8s.md

```diff
@@ -0,0 +1,313 @@
+---
+description: Setup a local Consul Service mesh to demonstrate how to use the ngrok Ingress Controller with Consul.
+---
+
+# Ingress to services in a Consul Service Mesh on Kubernetes
+------------
+
+:::tip TL;DR
+To use the ngrok Ingress Controller with Consul in a local demo environment:
+1. [Setup a local Consul Service Mesh on Kubernetes](#setup-a-consul-service-mesh-on-kubernetes)
+1. [Install the ngrok Ingress Controller](#install-the-ngrok-ingress-controller)
+1. [Install a sample application](#install-a-sample-application)
+1. [Configure Public Ingress for the sample application](#configure-public-ingress-for-the-sample-application)
+1. [Secure the app with Oauth](#secure-the-app-with-oauth)
+:::
+
+## Introduction
+
+The [ngrok Ingress Controller for Kubernetes](https://github.com/ngrok/kubernetes-ingress-controller) is our official open-source controller for adding public and secure ingress traffic to your k8s services. It works on any Cloud, local, or On-Prem Kubernetes cluster to provide ingress to your services no matter the network configuration, as long as it has outbound access to the ngrok service. This allows ngrok to be portable and work seamlessly across any type of infrastructure.
+
+[Consul](https://www.consul.io/) is a secure and resilient service mesh that provides service discovery, configuration, and segmentation functionality. Consul Connect provides service-to-service connection authorization and encryption using mutual TLS, automatically enabling TLS for all Connect services. Consul can be used with Kubernetes to provide a service mesh for your Kubernetes cluster.
+
+Together, Consul provides a robust and secure way for Services within a cluster to communicate, while ngrok can seamlessly and securely provide public ingress to those services. This guide will walk you through setting up a Consul Service Mesh on Kubernetes and then using the ngrok Ingress Controller to provide ingress to your services to illustrate how they can work together.
+
+:::caution This tutorial requires:
+1. A [free ngrok account](https://ngrok.com/signup).
+1. [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
+1. [Helm 3.0.0+](https://helm.sh/docs/intro/install/)
+1. A remote or local Kubernetes cluster with Consul installed _OR_ [minikube](https://minikube.sigs.k8s.io/docs/start/) to setup a demo cluster locally.
+:::
+
+## **Step 1**: Setup a local Consul Service Mesh on Kubernetes {#setup-a-consul-service-mesh-on-kubernetes}
+
+For this guide, we'll need access to a remote or local Kubernetes cluster with Consul installed. If you have an existing cluster with Consul setup, you can skip this step and proceed to [Step 2: Install the ngrok Ingress Controller](#install-the-ngrok-ingress-controller)
```

Comment:
> proceed to [Step 2: Install the ngrok Ingress Controller](#install-the-ngrok-ingress-controller)

proceed to [Step 2: Install the ngrok Ingress Controller](#install-the-ngrok-ingress-controller). <-- add a period here.